AudioTrack: Simplify Channel selection

fritsch opened the pull request:
All major Android devices have issues opening more "fine grain" audio channel configurations besides Stereo, 5.1 or .7.1 - use AE's feature to silently add additional channels so that e.g. 3.0, 4.0, 4.1, etc. are tunneled via 5.1, 7.1 and rest is sent via stereo.

The issue is here not kodi, but Android not being able to open the mask, throwing channels away while accepting the input without error.

Fixes:  #24546

fritsch committed:
AudioTrack: Simplify Channel selection

thexai commented on xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:
```diff
@@ -130,26 +130,18 @@ static int AEChannelMapToAUDIOTRACKChannelMask(CAEChannelInfo info)
+    return CJNIAudioFormat::CHANNEL_OUT_5POINT1;
 
-  for (unsigned int i = 0; i < info.Count(); i++)
-    atMask |= AEChannelToAUDIOTRACKChannel(info[i]);
```
Jenkins Android builds fails due new warning generated:

```
/home/jenkins/jenkins-root/workspace/android-arm-docker@2/xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp:93:12: warning: unused function 'AEChannelToAUDIOTRACKChannel' [-Wunused-function]
static int AEChannelToAUDIOTRACKChannel(AEChannel aeChannel)
           ^
1 warning generated.
```

fritsch replied:
I can remove this function as well. Will make a second commit though, as we need it back when we want to do it more properly again in the future.

fritsch committed:
AESinkAUDIOTRACK: Remove for now unneded AEChannelToAUDIOTRACKChannel method

AEChannelToAUDIOTRACKChannel is not used at the moment as we go with limited
Channelmasks Stereo, 5.1, 7.1. It is removed as separate commit as we will
need it back when Android devices appear that can do those advanced channel
layouts.

thexai reviewed (approved):
Tested on Shield and confirmed that now outputs muti-channel PCM with FLAC 4.0 and 4.1 sample files.

fritsch commented:
So - whatever comes up as regressions I will take up directly, which is why I did not wait longer with merging.
